compiler/compiler.go

```diff
@@ -53,6 +53,8 @@ func Compile(scope Scope, f *semantic.FunctionExpression, in semantic.MonoType)
 			if err := substituteTypes(subst, argT, mtyp); err != nil {
 				return nil, err
 			}
+		} else if !arg.Optional() {
```

Comment:
I found this omission when testing this because the complicated nature of the input for columns means that type inference can't really analyze the input. This is because the input for columns is an arbitrary object and the keys aren't statically defined.

This doesn't seem to affect any current code because all of our current functions can have their function specified in advance.